waku.md

```diff
@@ -56,64 +56,60 @@ Waku is a RLPx subprotocol called `waku` with version `0`. The version number co
 
 Using [Augmented Backus-Naur form (ABNF)](https://tools.ietf.org/html/rfc5234) we have the following format:
 
-<!-- TODO: This needs to be run through ABNF validator -->
-
 <!-- TODO: packet-format / packet rules, compact into one somehow? -->
 
 <!-- TOOD: ABNF floating point rep for pow? https://www.cs.cmu.edu/Groups/AI/html/cltl/clm/node19.html -->
 
 ```
-bytes            = [repeat] OCTET
-
 ; Packet codes 0 - 127 are reserved for Waku protocol
-packet-code      = 0 / 1 / 2 / 3 / [ x4-127 ]
-
-packet-format    = "[" packet-code packet-format "]"
-
-; packet-format needs to be paired with its corresponding packet-format
-packet           = "[" required-packet / [ optional-packet ] "]"
+packet-code     = 0 / 1 / 2 / 3 / [ x4-127 ]
 
-required-packet  = 0 status / 1 messages / 2 pow-requirement / 3 bloom-filter
-
-optional-packet  = 126 p2p-request / 127 p2p-message
-
-packet-format    = "[" packet-code packet-format "]"
-
-status           = "[" version pow-requirement [ bloom-filter ] [ light-node ] "]"
+status          = "[" version pow-requirement [ bloom-filter ] [ light-node ] "]"
 
 ; version is "an integer (as specified in RLP)"
-version          = DIGIT
-
-messages         = *waku-envelope
+version         = DIGIT
 
 ; pow is "a single floating point value of PoW. This value is the IEEE 754 binary representation of 64-bit floating point number. Values of qNAN, sNAN, INF and -INF are not allowed. Negative values are also not allowed."
-pow-requirement  = pow
+pow-requirement = pow
 
 ; bloom filter is "a byte array"
-bloom-filter     = bytes
-
-light-node       = BIT
+bloom-filter    = *OCTET
 
-p2p-request      = waku-envelope
+light-node      = BIT
 
-p2p-message      = waku-envelope
-
-whisper-envelope = "[" expiry ttl topic data nonce "]"
+waku-envelope   = "[" expiry ttl topic data nonce "]"
 
 ; 4 bytes (UNIX time in seconds)
-expiry           = bytes
+expiry          = 4*OCTET
 
 ; 4 bytes (time-to-live in seconds)
-ttl              = bytes
+ttl             = 4*OCTET
 
 ; 4 bytes of arbitrary data
-topic            = bytes
+topic           = 4*OCTET
 
 ; byte array of arbitrary size (contains encrypted message)
-data             = bytes
+data            = OCTET
 
 ; 8 bytes of arbitrary data (used for PoW calculation)
-nonce            = bytes
+nonce           = 8*OCTET
+
+messages        = *waku-envelope
+
+p2p-request     = waku-envelope
+
+p2p-message     = waku-envelope
```

Comment:
what about multiple envelopes here?

Reply:
do we allow multiple envelopes here?

Reply:
you reviewed the mailserver spec...

Reply:
> Historic messages MUST be sent to a peer as a packet with a P2P Message code (0x7f) followed by an array of Waku envelopes. It is incompatible with the original Whisper spec (EIP-627) because it allows only a single envelope, however, an array of envelopes is much more performant. In order to stay compatible with EIP-627, a peer receiving historic message MUST handle both cases.

you literally merged this

Reply:
fixed